src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuardsNew.java

```diff
@@ -0,0 +1,977 @@
+        for (final CitizenData citizen : getAssignedCitizen())
+        {
+            citizen.setJob(createJob(citizen));
+        }
+        this.markDirty();
+    }
+
+    /**
+     * Get the guard's {@link GuardTask}.
+     *
+     * @return The task of the guard.
+     */
+    public GuardTask getTask()
+    {
+        return this.task;
+    }
+
+    /**
+     * Set the guard's {@link GuardTask}.
+     *
+     * @param task The task to set.
+     */
+    public void setTask(final GuardTask task)
+    {
+        this.task = task;
+        this.markDirty();
+    }
+
+    public List<BlockPos> getPatrolTargets()
+    {
+        return new ArrayList<>(patrolTargets);
+    }
+
+    /**
+     * Get the guard's RetrieveOnLowHeath.
+     *
+     * @return if so.
+     */
+    public boolean shallRetrieveOnLowHealth()
+    {
+        return retrieveOnLowHealth;
+    }
+
+    /**
+     * Set the guard's RetrieveOnLowHealth.
+     *
+     * @param retrieve true if retrieve.
+     */
+    public void setRetrieveOnLowHealth(final boolean retrieve)
+    {
+        this.retrieveOnLowHealth = retrieve;
+    }
+
+    /**
+     * Get whether the guard should patrol manually.
+     *
+     * @return if so.
+     */
+    public boolean shallPatrolManually()
+    {
+        return patrolManually;
+    }
+
+    /**
+     * Set whether the guard should patrol manually.
+     *
+     * @param patrolManually true if manual.
+     */
+    public void setPatrolManually(final boolean patrolManually)
+    {
+        this.patrolManually = patrolManually;
+    }
+
+    /**
+     * Whether the player will assign guards manually or not.
+     *
+     * @return true if so
+     */
+    public boolean shallAssignManually()
+    {
+        return assignManually;
+    }
+
+    /**
+     * Set whether the player is assigning guards manually.
+     *
+     * @param assignManually true if so
+     */
+    public void setAssignManually(final boolean assignManually)
+    {
+        this.assignManually = assignManually;
+    }
+
+    /**
+     * Get the position the guard should guard.
+     *
+     * @return the {@link BlockPos} of the guard position.
+     */
+    public BlockPos getGuardPos()
+    {
+        return guardPos;
+    }
+
+    /**
+     * Set where the guard should guard.
+     *
+     * @param guardPos the {@link BlockPos} to guard.
+     */
+    public void setGuardPos(final BlockPos guardPos)
+    {
+        this.guardPos = guardPos;
+    }
+
+    /**
+     * Get the Map of mobs to attack.
+     *
+     * @return the map.
+     */
+    public List<MobEntryView> getMobsToAttack()
+    {
+        mobsToAttack.sort(Comparator.comparing(MobEntryView::getPriority, Comparator.reverseOrder()));
+        return mobsToAttack;
+    }
+
+    /**
+     * Set the Map of mobs to attack.
+     *
+     * @param list The new map.
+     */
+    public void setMobsToAttack(final List<MobEntryView> list)
+    {
+        this.mobsToAttack.clear();
+        this.mobsToAttack = list;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Store a copy of "list". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)